buffer/src/main/java/io/netty/buffer/AbstractReadableObject.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.buffer;
+
+import io.netty.util.IllegalReferenceCountException;
+import io.netty.util.ResourceLeak;
+import io.netty.util.ResourceLeakDetector;
+
+import java.io.IOException;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * A skeletal implementation of a {@link ReadableObject}.
+ */
+public abstract class AbstractReadableObject implements ReadableObject {
+    private static final ResourceLeakDetector<ReadableObject> leakDetector =
+            new ResourceLeakDetector<ReadableObject>(ReadableObject.class);
+
+    private final ResourceLeak leak;
+    private long readerPosition;
+
+    public AbstractReadableObject(long readerPosition) {
+        this.readerPosition = readerPosition;
+        leak = leakDetector.open(this);
```

Comment:
@Scottmitch absolutely ... I just wanted to point out where this came from in case there was a legitimate reason that `CompositeByteBuf` is different ... sounds like it's not.  I'll remove the leak detection code.